docs/xxcriurestorenonportablemode.md

```diff
@@ -0,0 +1,61 @@
+<!--
+* Copyright (c) 2017, 2023 IBM Corp. and others
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution and is available at
+* https://www.eclipse.org/legal/epl-2.0/ or the Apache
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -XX:\[+|-\]CRIURestoreNonPortableMode
+
+**(Linux&reg; x86, Linux on POWER LE&reg;, and Linux on IBM Z&reg; only)**
+
+This option enables and disables the JIT compiler from generating nonportable compiled code on restore.
+
+:fontawesome-solid-triangle-exclamation:{: .warn aria-hidden="true"} **Restrictions:**
+
+- This option takes effect only when [`-XX:+EnableCRIUSupport`](xxenablecriusupport.md) is enabled.
+- When this option is disabled, CRIUSECProvider is the only security provider in both checkpoint and restore.
+- JITServer is permanently disabled when this option is disabled.
+- The VM is limited to portable code for its entire lifetime when this option is disabled (`-XX:-CRIURestoreNonPortableMode`).
+
+
+## Syntax
+
+        -XX:[+|-]CRIURestoreNonPortableMode
+
+| Setting               | Effect  | Default                                                                            |
+|-----------------------|---------|:----------------------------------------------------------------------------------:|
+| `-XX:+CRIURestoreNonPortableMode` | Enable  | :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>    |
+| `-XX:-CRIURestoreNonPortableMode` | Disable |                                                                                 |
+
+
+## Explanation
+
+JIT compiled code uses the hardware on the system that it is running on, therefore it might include code that is not portable to a different system with different hardware capabilities. This nonportable compiled code might fail if the VM restores from the checkpoint files that include the incompatible nonportable code.
```

Comment:
I think it's the JIT compiler that's running, not the JIT-compiled code, and it's the CRIU checkpoint image that can include code that isn't portable, not the JIT-compiled code.
I also think this is one situation where "exploit" is  more accurate than "use".
The second sentence repeats itself somewhat.

I suggest something like "The JIT compiler can exploit hardware features of the system on which it is running, to generate compiled code. That code might therefore fail if it is included in a checkpoint image that you later restore on a different system. For example, the compiled code might try to exploit a hardware feature that doesn't exist on the new machine, then fail."